github/migrate_github_repository_webhook.go

```diff
@@ -0,0 +1,55 @@
+package github
+
+import (
+	"fmt"
+	"log"
+	"strings"
+
+	"github.com/hashicorp/terraform/terraform"
+)
+
+func resourceGithubRepositoryWebhookMigrateState(v int, is *terraform.InstanceState, meta interface{}) (*terraform.InstanceState, error) {
+	switch v {
+	case 0:
+		log.Println("[INFO] Found GitHub Repository Webhook State v0; migrating to v1")
+		return migrateGithubRepositoryWebhookStateV0toV1(is)
+	default:
+		return is, fmt.Errorf("Unexpected schema version: %d", v)
+	}
+}
+
+func migrateGithubRepositoryWebhookStateV0toV1(is *terraform.InstanceState) (*terraform.InstanceState, error) {
+	if is.Empty() {
+		log.Println("[DEBUG] Empty InstanceState; nothing to migrate.")
+		return is, nil
+	}
+
+	log.Printf("[DEBUG] GitHub Repository Webhook Attributes before migration: %#v", is.Attributes)
+
+	prefix := "configuration."
+
+	delete(is.Attributes, prefix+"%")
+
+	// Read & delete old keys
+	oldKeys := make(map[string]string, 0)
+	for k, v := range is.Attributes {
+		if strings.HasPrefix(k, prefix) {
+			oldKeys[k] = v
+
+			// Delete old keys
+			delete(is.Attributes, k)
+		}
+	}
+
+	// Write new keys
+	for k, v := range oldKeys {
```

Comment:
any chance this could be done all in the previous loop?

Reply:
That was actually the very first version of this PR 😃until I saw the test intermittently failing as the for loop kept ranging over some new elements which were just added.

Why? https://golang.org/ref/spec#For_range

> The iteration order over maps is not specified and is not guaranteed to be the same from one iteration to the next. If a map entry that has not yet been reached is removed during iteration, the corresponding iteration value will not be produced. **If a map entry is created during iteration, that entry may be produced during the iteration or may be skipped. The choice may vary for each entry created and from one iteration to the next.**

Short answer: No.

Feel free to have a play yourself: https://play.golang.org/p/Tm1FPyjwd0m 😉

Reply:
ahhhh yes I see now. 👍